src/lib.rs

```diff
@@ -122,6 +122,39 @@ pub fn decode<T: DeserializeOwned>(token: &str, key: &[u8], validation: &Validat
     Ok(TokenData { header: header, claims: decoded_claims })
 }
 
+/// Decode a token without any signature validation into a struct containing 2 fields: `claims` and `header`.
+///
+/// NOTE: Do not use this unless you know what you are doing! If the token's signature is invalid, it will *not* return an error.
+///
+/// ```rust,ignore
+/// #[macro_use]
+/// extern crate serde_derive;
+/// use jsonwebtoken::{dangerous_unsave_decode, Validation, Algorithm};
+///
+/// #[derive(Debug, Serialize, Deserialize)]
+/// struct Claims {
+///     sub: String,
+///     company: String
+/// }
+///
+/// let token = "a.jwt.token".to_string();
+/// // Claims is a struct that implements Deserialize
+/// let token_data = dangerous_unsafe_decode::<Claims>(&token, &Validation::new(Algorithm::HS256));
+/// ```
+pub fn dangerous_unsafe_decode<T: DeserializeOwned>(token: &str, validation: &Validation) -> Result<TokenData<T>> {
```

Comment:
if you just want to decode it, do you even care about other validation? It could just take the `token`

Reply:
@Keats I'll leave it up to you. I left these in to match 1.x/2.x behavior, but I could see all validations going away. I don't have a strong preference for either

Reply:
The only usecase for that fn is if you just want to see the data so doing the validation seems overkill. Remove it and I'll merge + release

Reply:
Alright, makes sense. Updated now. Thanks!